From 7a54da1fb3b713b7814f62debfedafd29fb97cef Mon Sep 17 00:00:00 2001
From: Fred Fan <r01011@freescale.com>
Date: Tue, 29 Jul 2008 18:06:56 +0800
Subject: [PATCH] ENGR00076031 Add support latest sdma script on i.MX32 ADS

apply the sdma script fo SS15 release and setup the entry address
of new sdma script. Merged from L26.22 for SONY.

Signed-off-by: Fred Fan <r01011@freescale.com>
---
 arch/arm/mach-mx3/devices.c                |   27 ++--
 arch/arm/mach-mx3/sdma_script_code_pass2.h |  298 ++++++++++++++++++----------
 arch/arm/plat-mxc/sdma/dma_sdma.c          |    5 +-
 3 files changed, 206 insertions(+), 124 deletions(-)

diff --git a/arch/arm/mach-mx3/devices.c b/arch/arm/mach-mx3/devices.c
index f2c727f..abf483e 100644
--- a/arch/arm/mach-mx3/devices.c
+++ b/arch/arm/mach-mx3/devices.c
@@ -76,45 +76,42 @@ void mxc_sdma_get_script_info(sdma_script_start_addrs * sdma_script_addr)
 		sdma_script_addr->mxc_sdma_uart_2_per_addr = -1;
 		sdma_script_addr->mxc_sdma_app_2_per_addr = -1;
 	} else {
-		sdma_script_addr->mxc_sdma_app_2_mcu_addr =
-		    app_2_mcu_patched_ADDR_2;
+		sdma_script_addr->mxc_sdma_app_2_mcu_addr = app_2_mcu_ADDR_2;
 		sdma_script_addr->mxc_sdma_ap_2_ap_addr = ap_2_ap_ADDR_2;
 		sdma_script_addr->mxc_sdma_ap_2_ap_fixed_addr =
 		    ap_2_ap_fixed_addr_ADDR_2;
 		sdma_script_addr->mxc_sdma_ap_2_bp_addr = ap_2_bp_ADDR_2;
+		sdma_script_addr->mxc_sdma_ap_2_ap_fixed_addr =
+		    ap_2_ap_fixed_addr_ADDR_2;
 		sdma_script_addr->mxc_sdma_bp_2_ap_addr = bp_2_ap_ADDR_2;
 		sdma_script_addr->mxc_sdma_loopback_on_dsp_side_addr = -1;
-		sdma_script_addr->mxc_sdma_mcu_2_app_addr =
-		    mcu_2_app_patched_ADDR_2;
-		sdma_script_addr->mxc_sdma_mcu_2_shp_addr =
-		    mcu_2_shp_patched_ADDR_2;
+		sdma_script_addr->mxc_sdma_mcu_2_app_addr = mcu_2_app_ADDR_2;
+		sdma_script_addr->mxc_sdma_mcu_2_shp_addr = mcu_2_shp_ADDR_2;
 		sdma_script_addr->mxc_sdma_mcu_interrupt_only_addr = -1;
-		sdma_script_addr->mxc_sdma_shp_2_mcu_addr =
-		    shp_2_mcu_patched_ADDR_2;
+		sdma_script_addr->mxc_sdma_shp_2_mcu_addr = shp_2_mcu_ADDR_2;
 		sdma_script_addr->mxc_sdma_start_addr =
 		    (unsigned short *)sdma_code_2;
 		sdma_script_addr->mxc_sdma_uartsh_2_mcu_addr =
-		    uartsh_2_mcu_patched_ADDR_2;
-		sdma_script_addr->mxc_sdma_uart_2_mcu_addr =
-		    uart_2_mcu_patched_ADDR_2;
+		    uartsh_2_mcu_ADDR_2;
+		sdma_script_addr->mxc_sdma_uart_2_mcu_addr = uart_2_mcu_ADDR_2;
 		sdma_script_addr->mxc_sdma_ram_code_size = RAM_CODE_SIZE_2;
 		sdma_script_addr->mxc_sdma_ram_code_start_addr =
 		    RAM_CODE_START_ADDR_2;
 		sdma_script_addr->mxc_sdma_dptc_dvfs_addr = -1;
 		sdma_script_addr->mxc_sdma_firi_2_mcu_addr = firi_2_mcu_ADDR_2;
 		sdma_script_addr->mxc_sdma_firi_2_per_addr = -1;
-		sdma_script_addr->mxc_sdma_mshc_2_mcu_addr = -1;
-		sdma_script_addr->mxc_sdma_per_2_app_addr = -1;
+		sdma_script_addr->mxc_sdma_mshc_2_mcu_addr = mshc_2_mcu_ADDR_2;
+		sdma_script_addr->mxc_sdma_per_2_app_addr = per_2_app_ADDR_2;
 		sdma_script_addr->mxc_sdma_per_2_firi_addr = -1;
 		sdma_script_addr->mxc_sdma_per_2_shp_addr = per_2_shp_ADDR_2;
 		sdma_script_addr->mxc_sdma_mcu_2_ata_addr = mcu_2_ata_ADDR_2;
 		sdma_script_addr->mxc_sdma_mcu_2_firi_addr = mcu_2_firi_ADDR_2;
-		sdma_script_addr->mxc_sdma_mcu_2_mshc_addr = -1;
+		sdma_script_addr->mxc_sdma_mcu_2_mshc_addr = mcu_2_mshc_ADDR_2;
 		sdma_script_addr->mxc_sdma_ata_2_mcu_addr = ata_2_mcu_ADDR_2;
 		sdma_script_addr->mxc_sdma_uartsh_2_per_addr = -1;
 		sdma_script_addr->mxc_sdma_shp_2_per_addr = shp_2_per_ADDR_2;
 		sdma_script_addr->mxc_sdma_uart_2_per_addr = -1;
-		sdma_script_addr->mxc_sdma_app_2_per_addr = -1;
+		sdma_script_addr->mxc_sdma_app_2_per_addr = app_2_per_ADDR_2;
 	}
 }
 
diff --git a/arch/arm/mach-mx3/sdma_script_code_pass2.h b/arch/arm/mach-mx3/sdma_script_code_pass2.h
index ab5ae08..85de716 100644
--- a/arch/arm/mach-mx3/sdma_script_code_pass2.h
+++ b/arch/arm/mach-mx3/sdma_script_code_pass2.h
@@ -3,13 +3,33 @@
  */
 
 /*
- * The code contained herein is licensed under the GNU General Public
- * License. You may obtain a copy of the GNU General Public License
- * Version 2 or later at the following locations:
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
  *
- * http://www.opensource.org/licenses/gpl-license.html
- * http://www.gnu.org/copyleft/gpl.html
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA  */
+
+/*!
+ * @file sdma_script_code.h
+ * @brief This file contains functions of SDMA scripts code initialization
+ *
+ * The file was generated automatically. Based on sdma scripts library.
+ *
+ * @ingroup SDMA
  */
+/*******************************************************************************
+
+			SDMA RELEASE LABEL: 	"SS15_MX31"
+
+*******************************************************************************/
 
 #ifndef __SDMA_SCRIPT_CODE_PASS2_H__
 #define __SDMA_SCRIPT_CODE_PASS2_H__
@@ -76,6 +96,13 @@
 #define app_2_mcu_patched_ADDR_2	6212
 #define app_2_mcu_patched_SIZE_2	104
 
+#undef  app_2_mcu_ADDR_2
+#undef  app_2_mcu_SIZE_2
+
+/*mapping the app_2_mcu start address to the patched(RAM)script start address*/
+#define app_2_mcu_ADDR_2		app_2_mcu_patched_ADDR_2
+#define app_2_mcu_SIZE_2		app_2_mcu_patched_SIZE_2
+
 #define app_2_per_ADDR_2	6316
 #define app_2_per_SIZE_2	105
 
@@ -91,43 +118,88 @@
 #define mcu_2_app_patched_ADDR_2	6735
 #define mcu_2_app_patched_SIZE_2	129
 
+#undef  mcu_2_app_ADDR_2
+#undef  mcu_2_app_SIZE_2
+
+/*mapping the mcu_2_app start address to the patched(RAM)script start address*/
+#define mcu_2_app_ADDR_2		mcu_2_app_patched_ADDR_2
+#define mcu_2_app_SIZE_2		mcu_2_app_patched_SIZE_2
+
 #define mcu_2_ata_ADDR_2	6864
 #define mcu_2_ata_SIZE_2	87
 
 #define mcu_2_firi_ADDR_2	6951
 #define mcu_2_firi_SIZE_2	77
 
-#define mcu_2_shp_patched_ADDR_2	7028
+#define mcu_2_mshc_ADDR_2	7028
+#define mcu_2_mshc_SIZE_2	48
+
+#define mcu_2_shp_patched_ADDR_2	7076
 #define mcu_2_shp_patched_SIZE_2	125
 
-#define per_2_app_ADDR_2	7153
+#undef  mcu_2_shp_ADDR_2
+#undef  mcu_2_shp_SIZE_2
+
+/*mapping the mcu_2_shp start address to the patched(RAM)script start address*/
+#define mcu_2_shp_ADDR_2		mcu_2_shp_patched_ADDR_2
+#define mcu_2_shp_SIZE_2		mcu_2_shp_patched_SIZE_2
+
+#define mshc_2_mcu_ADDR_2	7201
+#define mshc_2_mcu_SIZE_2	60
+
+#define per_2_app_ADDR_2	7261
 #define per_2_app_SIZE_2	131
 
-#define per_2_shp_ADDR_2	7284
+#define per_2_shp_ADDR_2	7392
 #define per_2_shp_SIZE_2	131
 
-#define shp_2_mcu_patched_ADDR_2	7415
+#define shp_2_mcu_patched_ADDR_2	7523
 #define shp_2_mcu_patched_SIZE_2	104
 
-#define shp_2_per_ADDR_2	7519
-#define shp_2_per_SIZE_2	106
+#undef  shp_2_mcu_ADDR_2
+#undef  shp_2_mcu_SIZE_2
+
+/*mapping the shp_2_mcu start address to the patched(RAM)script start address*/
+#define shp_2_mcu_ADDR_2		shp_2_mcu_patched_ADDR_2
+#define shp_2_mcu_SIZE_2		shp_2_mcu_patched_SIZE_2
 
-#define uart_2_mcu_patched_ADDR_2	7625
+#define shp_2_per_ADDR_2	7627
+#define shp_2_per_SIZE_2	109
+
+#define uart_2_mcu_patched_ADDR_2	7736
 #define uart_2_mcu_patched_SIZE_2	106
 
-#define uartsh_2_mcu_patched_ADDR_2	7731
+#undef  uart_2_mcu_ADDR_2
+#undef  uart_2_mcu_SIZE_2
+
+/*mapping the uart_2_mcu start address to the patched(RAM)script start address*/
+#define uart_2_mcu_ADDR_2		uart_2_mcu_patched_ADDR_2
+#define uart_2_mcu_SIZE_2		uart_2_mcu_patched_SIZE_2
+
+#define uartsh_2_mcu_patched_ADDR_2	7842
 #define uartsh_2_mcu_patched_SIZE_2	99
 
+#undef  uartsh_2_mcu_ADDR_2
+#undef  uartsh_2_mcu_SIZE_2
+
+/*
+ * mapping the uartsh_2_mcu start address to the patched(RAM)script
+ * start address
+ */
+#define uartsh_2_mcu_ADDR_2		uartsh_2_mcu_patched_ADDR_2
+#define uartsh_2_mcu_SIZE_2		uartsh_2_mcu_patched_SIZE_2
+
 /*!
 * SDMA RAM image start address and size
 */
 
 #define RAM_CODE_START_ADDR_2		6144
-#define RAM_CODE_SIZE_2			1686
+#define RAM_CODE_SIZE_2			1797
 
 /*!
 * Buffer that holds the SDMA RAM image
 */
+
 __attribute__ ((__aligned__(4)))
 #ifndef CONFIG_XIP_KERNEL
 const
@@ -244,105 +316,119 @@ static const short sdma_code_2[] = {
 	0x7e05, 0x7f01, 0x9b3c, 0x0007, 0x68cc, 0x9b68, 0x0007, 0x6a0c,
 	0x0454, 0x62c8, 0x7ef8, 0x5013, 0x52c0, 0x53c8, 0xc10d, 0x7dbd,
 	0x0200, 0x9b2d, 0x0200, 0x9b2a, 0xc19d, 0x0870, 0x0011, 0xc0ec,
+	0x7d29, 0x5010, 0x5ac0, 0x5ef8, 0xc0fe, 0x56f8, 0x7d02, 0x0200,
+	0x9b7b, 0x0870, 0x0011, 0x6d03, 0x0dff, 0x0511, 0x1dff, 0x05bc,
+	0x4d00, 0x7d12, 0x5228, 0x02b9, 0x4a00, 0x7c02, 0x0400, 0x9b8a,
+	0x620b, 0x7e06, 0x5a06, 0x7f06, 0x0000, 0x2504, 0x7d05, 0x9b8a,
+	0x0007, 0x680c, 0x0007, 0x0454, 0x5010, 0x52c0, 0xc10d, 0x7ddb,
+	0x0200, 0x9b7b, 0x0200, 0x9b77, 0xc19d, 0x0870, 0x0011, 0xc0ec,
 	0x7d76, 0x5010, 0x5ac0, 0x5bc8, 0x5ef8, 0xc0fe, 0x56f8, 0x7d02,
-	0x0200, 0x9b7b, 0x6d03, 0x0d03, 0x0512, 0x05bc, 0x0510, 0x5dd0,
+	0x0200, 0x9bab, 0x6d03, 0x0d03, 0x0512, 0x05bc, 0x0510, 0x5dd0,
 	0x0dff, 0x0511, 0x1dff, 0x05bc, 0x5df8, 0x4d00, 0x7d57, 0x0a70,
-	0x0211, 0x532a, 0x5212, 0x03b9, 0x4b00, 0x7c02, 0x0400, 0x9b8f,
+	0x0211, 0x532a, 0x5212, 0x03b9, 0x4b00, 0x7c02, 0x0400, 0x9bbf,
 	0x008f, 0x05d8, 0x7d01, 0x008d, 0x05a0, 0x5dda, 0x55d2, 0x4d00,
 	0x7d27, 0x4d02, 0x7d20, 0x4d01, 0x7d1a, 0x0a70, 0x0211, 0x52ea,
-	0x0260, 0x7d05, 0x6509, 0x7e25, 0x630a, 0x7e23, 0x9bb3, 0x630a,
+	0x0260, 0x7d05, 0x6509, 0x7e25, 0x630a, 0x7e23, 0x9be3, 0x630a,
 	0x7e20, 0x6509, 0x7e1e, 0x0512, 0x0512, 0x03ad, 0x5b06, 0x7f19,
-	0x2003, 0x4800, 0x7ced, 0x0a70, 0x0211, 0x5212, 0x9bce, 0x7802,
-	0x6309, 0x5b06, 0x9bcd, 0x0015, 0x7802, 0x630a, 0x5b06, 0x9bcd,
+	0x2003, 0x4800, 0x7ced, 0x0a70, 0x0211, 0x5212, 0x9bfe, 0x7802,
+	0x6309, 0x5b06, 0x9bfd, 0x0015, 0x7802, 0x630a, 0x5b06, 0x9bfd,
 	0x0015, 0x0015, 0x7802, 0x630b, 0x5b06, 0x7c03, 0x55da, 0x0000,
-	0x9b8d, 0x0007, 0x680c, 0x55d2, 0x4d00, 0x7d03, 0x4d02, 0x7d02,
-	0x9bdb, 0x0017, 0x0017, 0x55da, 0x009d, 0x55fa, 0x05a0, 0x08ff,
+	0x9bbd, 0x0007, 0x680c, 0x55d2, 0x4d00, 0x7d03, 0x4d02, 0x7d02,
+	0x9c0b, 0x0017, 0x0017, 0x55da, 0x009d, 0x55fa, 0x05a0, 0x08ff,
 	0x0011, 0x18ff, 0x0010, 0x04b8, 0x04ad, 0x0454, 0x0870, 0x0011,
-	0x5010, 0x52c0, 0x53c8, 0xc10d, 0x7d8e, 0x0200, 0x9b7b, 0x0200,
-	0x9b77, 0x0870, 0x0011, 0x5010, 0xc0ec, 0x7d7c, 0x5ac0, 0x5bc8,
-	0x5ef8, 0xc0fe, 0x56f8, 0x7d02, 0x0200, 0x9bf7, 0x6ed3, 0x6dc5,
-	0x0dff, 0x0511, 0x1dff, 0x05bc, 0x5df8, 0x4d00, 0x7d60, 0x0b70,
-	0x0311, 0x522b, 0x5313, 0x02b9, 0x4a00, 0x7c02, 0x0400, 0x9c07,
-	0x008f, 0x00d5, 0x7d01, 0x008d, 0x05a0, 0x5ddb, 0x0d03, 0x0512,
-	0x05bc, 0x0510, 0x5dd3, 0x4d00, 0x7d2c, 0x4d02, 0x7d24, 0x4d01,
-	0x7d1e, 0x59e3, 0x0b70, 0x0311, 0x53eb, 0x61c8, 0x7e2b, 0x62c8,
-	0x7e29, 0x65c8, 0x7e27, 0x0360, 0x7d03, 0x0112, 0x0112, 0x9c32,
-	0x0512, 0x0512, 0x0211, 0x02a9, 0x02ad, 0x6ac8, 0x7f1b, 0x2003,
-	0x4800, 0x7ceb, 0x0b70, 0x0311, 0x5313, 0x51e3, 0x9c4f, 0x7802,
-	0x62c8, 0x6ac8, 0x9c4e, 0x6dce, 0x0015, 0x7802, 0x62c8, 0x6ac8,
-	0x9c4e, 0x6dcf, 0x0015, 0x0015, 0x7801, 0x62d8, 0x7c03, 0x0000,
-	0x55db, 0x9c05, 0x0007, 0x68ff, 0x55d3, 0x4d00, 0x7d03, 0x4d02,
-	0x7d02, 0x9c5c, 0x0017, 0x0017, 0x55db, 0x009d, 0x55fb, 0x05a0,
-	0x08ff, 0x0011, 0x18ff, 0x0010, 0x04b8, 0x04ad, 0x0454, 0x62c8,
-	0x7ee9, 0x0870, 0x0011, 0x5010, 0x52c0, 0x53c8, 0xc10d, 0x7d87,
-	0x0200, 0x9bf7, 0x0200, 0x9bf4, 0xc19d, 0x0870, 0x0011, 0xc0ec,
-	0x7d7c, 0x5010, 0x5ac0, 0x5bc8, 0x5ef8, 0xc0fe, 0x56f8, 0x7d02,
-	0x0200, 0x9c7b, 0x6dc5, 0x0d03, 0x0512, 0x05bc, 0x0510, 0x5dd0,
-	0x0dff, 0x0511, 0x1dff, 0x05bc, 0x5df8, 0x4d00, 0x7d5d, 0x0a70,
-	0x0211, 0x532a, 0x5212, 0x03b9, 0x4b00, 0x7c02, 0x0400, 0x9c8f,
-	0x008f, 0x05d8, 0x7d01, 0x008d, 0x05a0, 0x5dda, 0x55d2, 0x4d00,
-	0x7d2c, 0x4d02, 0x7d24, 0x4d01, 0x7d1e, 0x59e2, 0x0a70, 0x0211,
-	0x52ea, 0x61c8, 0x7e2c, 0x63c8, 0x7e2a, 0x65c8, 0x7e28, 0x0260,
-	0x7d03, 0x0112, 0x0112, 0x9cb6, 0x0512, 0x0512, 0x0311, 0x03a9,
-	0x03ad, 0x5b06, 0x7f1c, 0x2003, 0x4800, 0x7ceb, 0x0a70, 0x0211,
-	0x5212, 0x51e2, 0x9cd4, 0x7802, 0x63c8, 0x5b06, 0x9cd3, 0x6dce,
-	0x0015, 0x7802, 0x63c8, 0x5b06, 0x9cd3, 0x6dcf, 0x0015, 0x0015,
-	0x7802, 0x63c8, 0x5b06, 0x7c03, 0x55da, 0x0000, 0x9c8d, 0x0007,
-	0x68ff, 0x55d2, 0x4d00, 0x7d03, 0x4d02, 0x7d02, 0x9ce1, 0x0017,
-	0x0017, 0x55da, 0x009d, 0x55fa, 0x05a0, 0x08ff, 0x0011, 0x18ff,
-	0x0010, 0x04b8, 0x04ad, 0x0454, 0x0870, 0x0011, 0x5010, 0x52c0,
-	0x53c8, 0xc10d, 0x7d88, 0x0200, 0x9c7b, 0x0200, 0x9c77, 0xc19d,
-	0x0870, 0x0011, 0xc0ec, 0x7d61, 0x5010, 0x5ac0, 0x5bc8, 0x5ef8,
-	0xc0fe, 0x56f8, 0x7d02, 0x0200, 0x9cfe, 0x6d07, 0x5df0, 0x0dff,
-	0x0511, 0x1dff, 0x05bc, 0x4d00, 0x7d45, 0x0a70, 0x0211, 0x532a,
-	0x5212, 0x03b9, 0x4b00, 0x7c04, 0x6a28, 0x7f3b, 0x0400, 0x9d0d,
-	0x008f, 0x05d8, 0x7d01, 0x008d, 0x05a0, 0x0b03, 0x0312, 0x03bc,
-	0x0310, 0x4b00, 0x7d1c, 0x4b02, 0x7d20, 0x4b01, 0x7d23, 0x0a70,
-	0x0211, 0x52ea, 0x5306, 0x7e25, 0x0260, 0x7d02, 0x0310, 0x0312,
-	0x6b09, 0x7f1f, 0x0312, 0x6b09, 0x7f1c, 0x0312, 0x6b09, 0x7f19,
-	0x2003, 0x4800, 0x7cef, 0x0a70, 0x0211, 0x5212, 0x9d4e, 0x0015,
-	0x0015, 0x7802, 0x5306, 0x6b0b, 0x9d4d, 0x0015, 0x7802, 0x5306,
-	0x6b0a, 0x9d4d, 0x7802, 0x5306, 0x6b09, 0x7c03, 0x6b28, 0x0000,
-	0x9d0b, 0xc77b, 0x6928, 0x7ffd, 0x0870, 0x0011, 0x5010, 0x52c0,
-	0x53c8, 0xc10d, 0x7da3, 0x0200, 0x9cfe, 0x0200, 0x9cfa, 0xc19d,
-	0x0870, 0x0011, 0xc0ec, 0x7d63, 0x5010, 0x5ac0, 0x5bc8, 0x5ef8,
-	0xc0fe, 0x56f8, 0x7d02, 0x0200, 0x9d66, 0x6dd7, 0x5df0, 0x0dff,
-	0x0511, 0x1dff, 0x05bc, 0x4d00, 0x7d47, 0x0a70, 0x0211, 0x532a,
-	0x5212, 0x03b9, 0x4b00, 0x7c04, 0x63c8, 0x7e3d, 0x0400, 0x9d75,
-	0x008f, 0x05d8, 0x7d01, 0x008d, 0x05a0, 0x0b03, 0x0312, 0x03bc,
-	0x0310, 0x4b00, 0x7d28, 0x4b02, 0x7d20, 0x4b01, 0x7d19, 0x6ddd,
-	0x0a70, 0x0211, 0x52ea, 0x5306, 0x7e26, 0x0260, 0x7d02, 0x0310,
-	0x0312, 0x6bc8, 0x7f20, 0x0312, 0x6bc8, 0x7f1d, 0x0312, 0x6bc8,
-	0x7f1a, 0x2003, 0x4800, 0x7cef, 0x0a70, 0x0211, 0x5212, 0x9db9,
-	0x6ddd, 0x7802, 0x5306, 0x6bc8, 0x9db8, 0x6dde, 0x0015, 0x7802,
-	0x5306, 0x6bc8, 0x9db8, 0x0015, 0x0015, 0x7802, 0x5306, 0x6bc8,
-	0x7c02, 0x0000, 0x9d73, 0xc777, 0x63c8, 0x7efd, 0x0870, 0x0011,
-	0x5010, 0x52c0, 0x53c8, 0xc10d, 0x7da1, 0x0200, 0x9d66, 0x0200,
-	0x9d62, 0x0870, 0x0011, 0x5010, 0xc0ec, 0x7d5c, 0x5ac0, 0x5bc8,
-	0x5ef8, 0xc0fe, 0x56f8, 0x7d02, 0x0200, 0x9dcf, 0x0b70, 0x0311,
-	0x6ec3, 0x6d07, 0x5df0, 0x0dff, 0x0511, 0x1dff, 0x05bc, 0x4d00,
-	0x7d3e, 0x522b, 0x02b9, 0x4a00, 0x7c04, 0x6a28, 0x7f34, 0x0400,
-	0x9de1, 0x028e, 0x1a94, 0x6ac3, 0x62c8, 0x0269, 0x7d1b, 0x1e94,
-	0x6ec3, 0x6ed3, 0x62c8, 0x0248, 0x6ac8, 0x2694, 0x6ec3, 0x62c8,
-	0x026e, 0x7d32, 0x6a09, 0x7f1f, 0x2501, 0x4d00, 0x7d20, 0x028e,
-	0x1a98, 0x6ac3, 0x62c8, 0x6ec3, 0x0260, 0x7df1, 0x6a28, 0x7f13,
-	0xc7af, 0x9e21, 0x6ee3, 0x008f, 0x2001, 0x00d5, 0x7d01, 0x008d,
-	0x05a0, 0x62c8, 0x026e, 0x7d18, 0x6a09, 0x7f05, 0x2001, 0x7cf9,
-	0x6a28, 0x0000, 0x9ddf, 0x0289, 0xc77b, 0x018a, 0x9e30, 0x6a28,
-	0x7ffa, 0x0b70, 0x0311, 0x5013, 0x52c0, 0x53c8, 0xc10d, 0x7da7,
-	0x0200, 0x9dcf, 0x0200, 0x9dcc, 0x6a28, 0x7fed, 0xc7af, 0x9e30,
-	0x0458, 0x0454, 0x9e21, 0xc19d, 0x0870, 0x0011, 0xc0ec, 0x7d55,
+	0x5010, 0x52c0, 0x53c8, 0xc10d, 0x7d8e, 0x0200, 0x9bab, 0x0200,
+	0x9ba7, 0xc19d, 0x0870, 0x0011, 0xc0ec, 0x7d35, 0x5010, 0x5ac0,
+	0x5ef8, 0xc0fe, 0x56f8, 0x7d02, 0x0200, 0x9c28, 0x0870, 0x0011,
+	0x6d07, 0x0dff, 0x0511, 0x1dff, 0x05bc, 0x4d00, 0x7d1c, 0x5228,
+	0x02b9, 0x4a00, 0x7c04, 0x6928, 0x7f0b, 0x0400, 0x9c37, 0x5206,
+	0x7e10, 0x6a0b, 0x6928, 0x7f04, 0x0000, 0x2504, 0x7d0c, 0x9c37,
+	0x0007, 0x680c, 0x680c, 0x6207, 0x6a07, 0x6a2b, 0x6a28, 0x0007,
+	0x680c, 0x0007, 0x0454, 0x6928, 0x7ff3, 0x5010, 0x52c0, 0xc10d,
+	0x7dcf, 0x0200, 0x9c28, 0x0200, 0x9c24, 0x0870, 0x0011, 0x5010,
+	0xc0ec, 0x7d7c, 0x5ac0, 0x5bc8, 0x5ef8, 0xc0fe, 0x56f8, 0x7d02,
+	0x0200, 0x9c63, 0x6ed3, 0x6dc5, 0x0dff, 0x0511, 0x1dff, 0x05bc,
+	0x5df8, 0x4d00, 0x7d60, 0x0b70, 0x0311, 0x522b, 0x5313, 0x02b9,
+	0x4a00, 0x7c02, 0x0400, 0x9c73, 0x008f, 0x00d5, 0x7d01, 0x008d,
+	0x05a0, 0x5ddb, 0x0d03, 0x0512, 0x05bc, 0x0510, 0x5dd3, 0x4d00,
+	0x7d2c, 0x4d02, 0x7d24, 0x4d01, 0x7d1e, 0x59e3, 0x0b70, 0x0311,
+	0x53eb, 0x61c8, 0x7e2b, 0x62c8, 0x7e29, 0x65c8, 0x7e27, 0x0360,
+	0x7d03, 0x0112, 0x0112, 0x9c9e, 0x0512, 0x0512, 0x0211, 0x02a9,
+	0x02ad, 0x6ac8, 0x7f1b, 0x2003, 0x4800, 0x7ceb, 0x0b70, 0x0311,
+	0x5313, 0x51e3, 0x9cbb, 0x7802, 0x62c8, 0x6ac8, 0x9cba, 0x6dce,
+	0x0015, 0x7802, 0x62c8, 0x6ac8, 0x9cba, 0x6dcf, 0x0015, 0x0015,
+	0x7801, 0x62d8, 0x7c03, 0x0000, 0x55db, 0x9c71, 0x0007, 0x68ff,
+	0x55d3, 0x4d00, 0x7d03, 0x4d02, 0x7d02, 0x9cc8, 0x0017, 0x0017,
+	0x55db, 0x009d, 0x55fb, 0x05a0, 0x08ff, 0x0011, 0x18ff, 0x0010,
+	0x04b8, 0x04ad, 0x0454, 0x62c8, 0x7ee9, 0x0870, 0x0011, 0x5010,
+	0x52c0, 0x53c8, 0xc10d, 0x7d87, 0x0200, 0x9c63, 0x0200, 0x9c60,
+	0xc19d, 0x0870, 0x0011, 0xc0ec, 0x7d7c, 0x5010, 0x5ac0, 0x5bc8,
+	0x5ef8, 0xc0fe, 0x56f8, 0x7d02, 0x0200, 0x9ce7, 0x6dc5, 0x0d03,
+	0x0512, 0x05bc, 0x0510, 0x5dd0, 0x0dff, 0x0511, 0x1dff, 0x05bc,
+	0x5df8, 0x4d00, 0x7d5d, 0x0a70, 0x0211, 0x532a, 0x5212, 0x03b9,
+	0x4b00, 0x7c02, 0x0400, 0x9cfb, 0x008f, 0x05d8, 0x7d01, 0x008d,
+	0x05a0, 0x5dda, 0x55d2, 0x4d00, 0x7d2c, 0x4d02, 0x7d24, 0x4d01,
+	0x7d1e, 0x59e2, 0x0a70, 0x0211, 0x52ea, 0x61c8, 0x7e2c, 0x63c8,
+	0x7e2a, 0x65c8, 0x7e28, 0x0260, 0x7d03, 0x0112, 0x0112, 0x9d22,
+	0x0512, 0x0512, 0x0311, 0x03a9, 0x03ad, 0x5b06, 0x7f1c, 0x2003,
+	0x4800, 0x7ceb, 0x0a70, 0x0211, 0x5212, 0x51e2, 0x9d40, 0x7802,
+	0x63c8, 0x5b06, 0x9d3f, 0x6dce, 0x0015, 0x7802, 0x63c8, 0x5b06,
+	0x9d3f, 0x6dcf, 0x0015, 0x0015, 0x7802, 0x63c8, 0x5b06, 0x7c03,
+	0x55da, 0x0000, 0x9cf9, 0x0007, 0x68ff, 0x55d2, 0x4d00, 0x7d03,
+	0x4d02, 0x7d02, 0x9d4d, 0x0017, 0x0017, 0x55da, 0x009d, 0x55fa,
+	0x05a0, 0x08ff, 0x0011, 0x18ff, 0x0010, 0x04b8, 0x04ad, 0x0454,
+	0x0870, 0x0011, 0x5010, 0x52c0, 0x53c8, 0xc10d, 0x7d88, 0x0200,
+	0x9ce7, 0x0200, 0x9ce3, 0xc19d, 0x0870, 0x0011, 0xc0ec, 0x7d61,
+	0x5010, 0x5ac0, 0x5bc8, 0x5ef8, 0xc0fe, 0x56f8, 0x7d02, 0x0200,
+	0x9d6a, 0x6d07, 0x5df0, 0x0dff, 0x0511, 0x1dff, 0x05bc, 0x4d00,
+	0x7d45, 0x0a70, 0x0211, 0x532a, 0x5212, 0x03b9, 0x4b00, 0x7c04,
+	0x6a28, 0x7f3b, 0x0400, 0x9d79, 0x008f, 0x05d8, 0x7d01, 0x008d,
+	0x05a0, 0x0b03, 0x0312, 0x03bc, 0x0310, 0x4b00, 0x7d1c, 0x4b02,
+	0x7d20, 0x4b01, 0x7d23, 0x0a70, 0x0211, 0x52ea, 0x5306, 0x7e25,
+	0x0260, 0x7d02, 0x0310, 0x0312, 0x6b09, 0x7f1f, 0x0312, 0x6b09,
+	0x7f1c, 0x0312, 0x6b09, 0x7f19, 0x2003, 0x4800, 0x7cef, 0x0a70,
+	0x0211, 0x5212, 0x9dba, 0x0015, 0x0015, 0x7802, 0x5306, 0x6b0b,
+	0x9db9, 0x0015, 0x7802, 0x5306, 0x6b0a, 0x9db9, 0x7802, 0x5306,
+	0x6b09, 0x7c03, 0x6b28, 0x0000, 0x9d77, 0xc77b, 0x6928, 0x7ffd,
+	0x0870, 0x0011, 0x5010, 0x52c0, 0x53c8, 0xc10d, 0x7da3, 0x0200,
+	0x9d6a, 0x0200, 0x9d66, 0xc19d, 0x0870, 0x0011, 0xc0ec, 0x7d60,
 	0x5010, 0x5ac0, 0x5bc8, 0x5ef8, 0xc0fe, 0x56f8, 0x7d02, 0x0200,
-	0x9e3a, 0x0b70, 0x0311, 0x6d07, 0x5df0, 0x0dff, 0x0511, 0x1dff,
-	0x05bc, 0x4d00, 0x7d37, 0x522b, 0x02b9, 0x4a00, 0x7c04, 0x6928,
-	0x7f2d, 0x0400, 0x9e4b, 0x028e, 0x1a94, 0x5202, 0x0269, 0x7d16,
-	0x1e94, 0x5206, 0x0248, 0x5a06, 0x2694, 0x5206, 0x026e, 0x7d2f,
-	0x6a09, 0x7f1c, 0x2501, 0x4d00, 0x7d1d, 0x028e, 0x1a98, 0x5202,
-	0x0260, 0x7df3, 0x6a28, 0x7f12, 0xc7af, 0x9e84, 0x008f, 0x2001,
-	0x00d5, 0x7d01, 0x008d, 0x05a0, 0x5206, 0x026e, 0x7d18, 0x6a09,
-	0x7f05, 0x2001, 0x7cf9, 0x6a28, 0x0000, 0x9e49, 0x0289, 0xc77b,
-	0x018a, 0x9e93, 0x6928, 0x7ffa, 0x0b70, 0x0311, 0x5013, 0x52c0,
-	0x53c8, 0xc10d, 0x7daf, 0x0200, 0x9e3a, 0x0200, 0x9e36, 0x6a28,
-	0x7fed, 0xc7af, 0x9e93, 0x0458, 0x0454, 0x9e84
+	0x9dd2, 0x6dd7, 0x5df0, 0x0dff, 0x0511, 0x1dff, 0x05bc, 0x4d00,
+	0x7d46, 0x0a70, 0x0211, 0x532a, 0x5212, 0x03b9, 0x4b00, 0x7c02,
+	0x0400, 0x9de1, 0x008f, 0x05d8, 0x7d01, 0x008d, 0x05a0, 0x0b03,
+	0x0312, 0x03bc, 0x0310, 0x4b00, 0x7d28, 0x4b02, 0x7d20, 0x4b01,
+	0x7d19, 0x6ddd, 0x0a70, 0x0211, 0x52ea, 0x5306, 0x7e27, 0x0260,
+	0x7d02, 0x0310, 0x0312, 0x6bc8, 0x7f21, 0x0312, 0x6bc8, 0x7f1e,
+	0x0312, 0x6bc8, 0x7f1b, 0x2003, 0x4800, 0x7cef, 0x0a70, 0x0211,
+	0x5212, 0x9e23, 0x6ddd, 0x7802, 0x5306, 0x6bc8, 0x9e22, 0x6dde,
+	0x0015, 0x7802, 0x5306, 0x6bc8, 0x9e22, 0x0015, 0x0015, 0x7802,
+	0x5306, 0x6bc8, 0x7c03, 0x0000, 0xde32, 0x9ddf, 0xc777, 0x0870,
+	0x0011, 0x5010, 0x52c0, 0x53c8, 0xc10d, 0x7da4, 0x0200, 0x9dd2,
+	0x0200, 0x9dce, 0x63ff, 0x0368, 0x7d02, 0x0369, 0x7def, 0x0006,
+	0x0870, 0x0011, 0x5010, 0xc0ec, 0x7d5c, 0x5ac0, 0x5bc8, 0x5ef8,
+	0xc0fe, 0x56f8, 0x7d02, 0x0200, 0x9e3e, 0x0b70, 0x0311, 0x6ec3,
+	0x6d07, 0x5df0, 0x0dff, 0x0511, 0x1dff, 0x05bc, 0x4d00, 0x7d3e,
+	0x522b, 0x02b9, 0x4a00, 0x7c04, 0x6a28, 0x7f34, 0x0400, 0x9e50,
+	0x028e, 0x1a94, 0x6ac3, 0x62c8, 0x0269, 0x7d1b, 0x1e94, 0x6ec3,
+	0x6ed3, 0x62c8, 0x0248, 0x6ac8, 0x2694, 0x6ec3, 0x62c8, 0x026e,
+	0x7d32, 0x6a09, 0x7f1f, 0x2501, 0x4d00, 0x7d20, 0x028e, 0x1a98,
+	0x6ac3, 0x62c8, 0x6ec3, 0x0260, 0x7df1, 0x6a28, 0x7f13, 0xc7af,
+	0x9e90, 0x6ee3, 0x008f, 0x2001, 0x00d5, 0x7d01, 0x008d, 0x05a0,
+	0x62c8, 0x026e, 0x7d18, 0x6a09, 0x7f05, 0x2001, 0x7cf9, 0x6a28,
+	0x0000, 0x9e4e, 0x0289, 0xc77b, 0x018a, 0x9e9f, 0x6a28, 0x7ffa,
+	0x0b70, 0x0311, 0x5013, 0x52c0, 0x53c8, 0xc10d, 0x7da7, 0x0200,
+	0x9e3e, 0x0200, 0x9e3b, 0x6a28, 0x7fed, 0xc7af, 0x9e9f, 0x0458,
+	0x0454, 0x9e90, 0xc19d, 0x0870, 0x0011, 0xc0ec, 0x7d55, 0x5010,
+	0x5ac0, 0x5bc8, 0x5ef8, 0xc0fe, 0x56f8, 0x7d02, 0x0200, 0x9ea9,
+	0x0b70, 0x0311, 0x6d07, 0x5df0, 0x0dff, 0x0511, 0x1dff, 0x05bc,
+	0x4d00, 0x7d37, 0x522b, 0x02b9, 0x4a00, 0x7c04, 0x6928, 0x7f2d,
+	0x0400, 0x9eba, 0x028e, 0x1a94, 0x5202, 0x0269, 0x7d16, 0x1e94,
+	0x5206, 0x0248, 0x5a06, 0x2694, 0x5206, 0x026e, 0x7d2f, 0x6a09,
+	0x7f1c, 0x2501, 0x4d00, 0x7d1d, 0x028e, 0x1a98, 0x5202, 0x0260,
+	0x7df3, 0x6a28, 0x7f12, 0xc7af, 0x9ef3, 0x008f, 0x2001, 0x00d5,
+	0x7d01, 0x008d, 0x05a0, 0x5206, 0x026e, 0x7d18, 0x6a09, 0x7f05,
+	0x2001, 0x7cf9, 0x6a28, 0x0000, 0x9eb8, 0x0289, 0xc77b, 0x018a,
+	0x9f02, 0x6928, 0x7ffa, 0x0b70, 0x0311, 0x5013, 0x52c0, 0x53c8,
+	0xc10d, 0x7daf, 0x0200, 0x9ea9, 0x0200, 0x9ea5, 0x6a28, 0x7fed,
+	0xc7af, 0x9f02, 0x0458, 0x0454, 0x9ef3
 };
 #endif
diff --git a/arch/arm/plat-mxc/sdma/dma_sdma.c b/arch/arm/plat-mxc/sdma/dma_sdma.c
index 8c9835e..8bef5e9 100644
--- a/arch/arm/plat-mxc/sdma/dma_sdma.c
+++ b/arch/arm/plat-mxc/sdma/dma_sdma.c
@@ -295,12 +295,11 @@ int mxc_dma_config(int channel_num, mxc_dma_requestbuf_t * dma_buf,
 				chnl_param.transfer_type = emi_2_dsp;
 			}
 		} else if (chnl_param.peripheral_type == FIFO_MEMORY) {
-			if (mode == MXC_DMA_MODE_READ) {
+			if (mode == MXC_DMA_MODE_READ)
 				chnl_param.per_address = MXC_FIFO_MEM_SRC_FIXED;
-			} else {
+			else
 				chnl_param.per_address =
 				    MXC_FIFO_MEM_DEST_FIXED;
-			}
 		} else {
 			if (mode == MXC_DMA_MODE_READ) {
 				chnl_param.transfer_type = per_2_emi;
-- 
1.5.4.4

